server/budgettracker.business/Api/BudgetApi.cs

```diff
@@ -113,15 +113,21 @@ public async Task<ApiResponse> DeleteBudgets(ApiRequest request)
 
         public async Task<ApiResponse> GetBudget(ApiRequest request)
         {
-            await Authenticate(request);
+            User user = await Authenticate(request);
 
             GetBudgetArgumentApiContract getBudget = request.Arguments<GetBudgetArgumentApiContract>();
 
             ApiResponse response = new ApiResponse();
 
             try
             {
-                Budget retrievedBudget = await _budgetRepository.GetBudget(getBudget.BudgetValues.Id);                
+                Budget retrievedBudget = await _budgetRepository.GetBudget(getBudget.BudgetValues.Id);
+
+                if (retrievedBudget.Owner.Id != user.Id) 
+                {
+                    throw new RepositoryException("Retrieve a budget with invalid user Id.");
```

Comment:
When you are fixing Merge conflicts, consider changing this to something more official. Like `The authenticated user does not have permission to view this budget.`

What you have sounds like you're telling the user to retrieve the budget... which is what we're throwing an error about lol.

Reply:
True true, will update